lib/galaxy/objectstore/s3.py

```diff
@@ -408,20 +438,20 @@ def file_ready(self, obj, **kwargs):
         return False
 
     def exists(self, obj, **kwargs):
-        in_cache = in_s3 = False
+        in_cache = False
         rel_path = self._construct_path(obj, **kwargs)
 
         # Check cache
         if self._in_cache(rel_path):
             in_cache = True
         # Check S3
-        in_s3 = self._key_exists(rel_path)
+        in_cloud = self._key_exists(rel_path)
```

Comment:
if this is now a generic implementation of any-cloud-based object store the module shouldn't be called `s3.py` and many other changes in this code should happen.